dotcom-rendering/src/lib/useCommentCount.ts

```diff
@@ -4,26 +4,38 @@ import { useApi } from './useApi';
 
 type CommentCounts = Record<string, number>;
 
-const getInitialSetFromDOMAttribute = (attribute: string) =>
+const attribute = 'data-discussion-id';
+
+const getIdsFromDOM = () =>
 	[...document.querySelectorAll(`[${attribute}]`)]
 		.map((element) => element.getAttribute(attribute))
 		.filter(isNonNullable);
 
-const uniqueDiscussionIds = new Set<string>(
-	isServer ? [] : getInitialSetFromDOMAttribute('data-discussion-id'),
-);
+/** Ensure that we reduce the number of requests to get comment counts */
+export const addDiscussionIds = (ids: string[]): void => {
+	for (const id of ids.concat(getIdsFromDOM())) {
+		uniqueDiscussionIds.add(id);
+	}
+};
+
+const makeHTTPS = [/^http:\/\//, 'https://'] as const;
```

Comment:
I’m puzzled as to why the `discussionApiUrl` would come as naked HTTP and maybe this is not something that should be fixed so deeply!

Reply:
Resolved in #8619